provisioning/device/src/DeviceProvisioningClientException.cs

```diff
@@ -2,107 +2,62 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Net;
 using System.Runtime.Serialization;
 
 namespace Microsoft.Azure.Devices.Provisioning.Client
 {
     /// <summary>
-    /// Represents errors reported by the Provisioning Transport Handlers.
+    /// The exception that is thrown when an error occurs during device provisioning client operation.
     /// </summary>
-    public class ProvisioningTransportException : Exception
+    public class DeviceProvisioningClientException : Exception
```

Comment:
Do we like the naming here `DeviceProvisioningClientException`? Maybe I should change this to `ProvisioningDeviceClientException` to align the naming style for dps device & service clients?

Reply:
Can modules register with DPS, @timtay-microsoft / @abhipsaMisra ?